python/lsst/ts/hvac/mqtt_info_reader.py

```diff
@@ -0,0 +1,467 @@
+
+        Returns
+        -------
+        lower_limit: `float`
+            The lower limit
+        upper_limit: `float`
+            The upper limit
+
+        Raises
+        ------
+        ValueError
+            In case an unknown string pattern is found in the limits column.
+
+        """
+        lower_limit = None
+        upper_limit = None
+
+        match = re.match(
+            r"^(-?\d+)(/| a |% a |°C a | bar a |%RH a )(-?\d+)(%|°C| bar| hr|%RH)?$",
+            limits_string,
+        )
+        if match:
+            lower_limit = float(match.group(1))
+            upper_limit = float(match.group(3))
+        elif re.match(r"^\d$", limits_string):
+            lower_limit = 0
+            upper_limit = 100
+        elif limits_string == "1,2,3,4,5,6,7,8":
+            lower_limit = 1
+            upper_limit = 8
+        elif limits_string == "1,2,3,4,5,6":
+            lower_limit = 1
+            upper_limit = 6
+        elif limits_string in ["true o false", "-", ""]:
+            # ignore because there really are no lower and upper limits
+            pass
+        else:
+            raise ValueError(f"Couldn't match limits_string {limits_string}")
+
+        return lower_limit, upper_limit
+
+    def extract_topic_and_item(self, topic_and_item):
+        """Extract the generic topic, representing a HVAC subsystem, and item,
+         representing a published value of the subsystem, from a string.
+
+        This method searches for the last occurrence of that forward slash and
+        will return the part before the slash as topic and after the slash as
+        item.
+
+        Parameters
+        ----------
+        topic_and_item: `str`
+            The string to extract the generic topic and item from.
+
+        Returns
+        -------
+        topic: `str`
+            Defined as the part before the last forward slash.
+        item: `str`:
+             Defined as the part after the last forward slash.
+
+        Raises
+        ------
+        ValueError
+            In case no forward slash is found.
+        """
+        # This throws a ValueError in case no forward slash is found.
+        topic, item = topic_and_item.rsplit("/", 1)
+        return topic, item
+
+    def _generic_collect_topics_and_items(
+        self, topic_and_item, topic_type, idl_type, unit, limits, topics, items
+    ):
+        """Collect XML topics and items from a row read from the CSV or JSON
+        file with the Rubin Observatory HVAC system information sent by
+        DatControl.
+
+        Parameters
+        ----------
+        topic_and_item: `str`
+            The topic and item to parse, for instance
+
+                "LSST/PISO02/CRACK01/ESTADO_FUNCIONAMIENTO"
+
+            would be the topic "LSST/PISO02/CRACK01" and the item
+            "ESTADO_FUNCIONAMIENTO"
+        topic_type: `str`
+            Indicates whether the topic is a telemetry topic (READ) or a
+            command topic (WRITE).
```

Comment:
I gently suggest you use a string-valued enum, e.g.:
```
class TopicType(str, enum.Enum):
    READ = "READ"
    WRITE = "WRITE"
```
then your code can use enum constants and catch typos and other invalid values. By inheriting from str first, the enum values can be used directly as strings (a trick I just learned). If you don't care about that then don't inherit from str.

Reply:
That's a really neat trick. Thanks and done.